### PR TITLE
Fixes some divisions by 0

### LINKED ISSUE #9442: Division by zero - data_huds.dm,411
```
runtime error: Division by zero
 - proc name: diag hud set mechcell (/obj/mecha/proc/diag_hud_set_mechcell)
 -   source file: code/game/data_huds.dm,411
 -   usr: null
 -   src: SLIMER THE GOD (/obj/mecha/combat/durand)
 -   src.loc: the grass patch (175,82,9) (/turf/open/floor/grass)
 -   call stack:
 - SLIMER THE GOD (/obj/mecha/combat/durand): diag hud set mechcell()
 - SLIMER THE GOD (/obj/mecha/combat/durand): process(2)
 - SLIMER THE GOD (/obj/mecha/combat/durand): process(2)
 - Objects (/datum/controller/subsystem/processing/obj): fire(0)
 - Objects (/datum/controller/subsystem/processing/obj): ignite(0)
 - Master (/datum/controller/master): RunQueue()
 - Master (/datum/controller/master): Loop()
 - Master (/datum/controller/master): StartProcessing(0)
```

### LINKED ISSUE #9443: Division by zero - cell.dm
image

yeah, it's simple.

### LINKED ISSUE #9445: Division by zero - spacevine.dm,488
```
runtime error: Division by zero
 - proc name: process (/datum/spacevine_controller/process)
 -   source file: spacevine.dm,488
 -   usr: null
 -   src: /datum/spacevine_controller (/datum/spacevine_controller)
 -   call stack:
 - /datum/spacevine_controller (/datum/spacevine_controller): process(2)
 - Objects (/datum/controller/subsystem/processing/obj): fire(1)
 - Objects (/datum/controller/subsystem/processing/obj): ignite(1)
 - Master (/datum/controller/master): RunQueue()
 - Master (/datum/controller/master): Loop()
 - Master (/datum/controller/master): StartProcessing(0)
```

relevant line:
image

### LINKED ISSUE #9447: Division by zero - battle_royale.dm,249
```
runtime error: Division by zero
 - proc name: process (/datum/battle_royale_controller/process)
 -   source file: battle_royale.dm,249
 -   usr: null
 -   src: /datum/battle_royale_controlle... (/datum/battle_royale_controller)
 -   call stack:
 - /datum/battle_royale_controlle... (/datum/battle_royale_controller): process(1)
 - Processing (/datum/controller/subsystem/processing): fire(0)
 - Processing (/datum/controller/subsystem/processing): ignite(0)
 - Master (/datum/controller/master): RunQueue()
 - Master (/datum/controller/master): Loop()
 - Master (/datum/controller/master): StartProcessing(0)
 ```
image

### LINKED ISSUE #9448: Division by zero - energy.dm,78
```
runtime error: Division by zero
 - proc name: fire sounds (/obj/item/gun/energy/fire_sounds)
 -   source file: energy.dm,78
 -   usr: [REDACTED] (/mob/living/carbon/human)
 -   src: the pulse rifle (Wielded) (/obj/item/gun/energy/pulse/loyalpin)
 -   usr.loc: the clockwork floor (196,76,1) (/turf/open/floor/clockwork)
 -   src.loc: [REDACTED] (/mob/living/carbon/human)
 -   call stack:
 - the pulse rifle (Wielded) (/obj/item/gun/energy/pulse/loyalpin): fire sounds()
 - the pulse rifle (Wielded) (/obj/item/gun/energy/pulse/loyalpin): shoot live shot([REDACTED] (/mob/living/carbon/human), 0, the clockwork floor (193,79,1) (/turf/open/floor/clockwork), 1)
 - the pulse rifle (Wielded) (/obj/item/gun/energy/pulse/loyalpin): process fire(the clockwork floor (193,79,1) (/turf/open/floor/clockwork), [REDACTED] (/mob/living/carbon/human), 1, "icon-x=7;icon-y=25;left=1;butt...", "", 0, 0)
 - the pulse rifle (Wielded) (/obj/item/gun/energy/pulse/loyalpin): process fire(the clockwork floor (193,79,1) (/turf/open/floor/clockwork), [REDACTED] (/mob/living/carbon/human), 1, "icon-x=7;icon-y=25;left=1;butt...", "", 0, null)
 - the pulse rifle (Wielded) (/obj/item/gun/energy/pulse/loyalpin): afterattack(the clockwork floor (193,79,1) (/turf/open/floor/clockwork), [REDACTED] (/mob/living/carbon/human), 0, "icon-x=7;icon-y=25;left=1;butt...", null)
 - [REDACTED] (/mob/living/carbon/human): ClickOn(the clockwork floor (193,79,1) (/turf/open/floor/clockwork), "icon-x=7;icon-y=25;left=1;butt...")
 - the clockwork floor (193,79,1) (/turf/open/floor/clockwork): Click(the clockwork floor (193,79,1) (/turf/open/floor/clockwork), "mapwindow.map", "icon-x=7;icon-y=25;left=1;butt...")
 - [REDACTED_CLI] (/client): Click(the clockwork floor (193,79,1) (/turf/open/floor/clockwork), the clockwork floor (193,79,1) (/turf/open/floor/clockwork), "mapwindow.map", "icon-x=7;icon-y=25;left=1;butt...")
```

### LINKED ISSUE #9449: Division by zero - butchering.dm,56
```

runtime error: Division by zero
 - proc name: startNeckSlice (/datum/component/butchering/proc/startNeckSlice)
 -   source file: butchering.dm,56
 -   usr: [REDACTED_MOB] (/mob/living/carbon/human)
 -   src: /datum/component/butchering (/datum/component/butchering)
 -   usr.loc: the floor (75,170,2) (/turf/open/floor/wood)
 -   call stack:
 - /datum/component/butchering (/datum/component/butchering): startNeckSlice(the chainsaw (Wielded) (/obj/item/chainsaw), [REDACTED_MOB_SECOND] (/mob/living/carbon/human), [REDACTED_MOB] (/mob/living/carbon/human))
 - world: ImmediateInvokeAsync(/datum/component/butchering (/datum/component/butchering), /datum/component/butchering/pr... (/datum/component/butchering/proc/startNeckSlice), the chainsaw (Wielded) (/obj/item/chainsaw), [REDACTED_MOB_SECOND] (/mob/living/carbon/human), [REDACTED_MOB] (/mob/living/carbon/human))
 - /datum/component/butchering (/datum/component/butchering): onItemAttack(the chainsaw (Wielded) (/obj/item/chainsaw), [REDACTED_MOB_SECOND] (/mob/living/carbon/human), [REDACTED_MOB] (/mob/living/carbon/human))
 - the chainsaw (Wielded) (/obj/item/chainsaw):  SendSignal("item_attack", /list (/list))
 - the chainsaw (Wielded) (/obj/item/chainsaw): attack([REDACTED_MOB_SECOND] (/mob/living/carbon/human), [REDACTED_MOB] (/mob/living/carbon/human))
 - [REDACTED_MOB_SECOND] (/mob/living/carbon/human): attackby(the chainsaw (Wielded) (/obj/item/chainsaw), [REDACTED_MOB] (/mob/living/carbon/human), "icon-x=15;icon-y=18;vis-x=15;v...")
 - [REDACTED_MOB_SECOND] (/mob/living/carbon/human): attackby(the chainsaw (Wielded) (/obj/item/chainsaw), [REDACTED_MOB] (/mob/living/carbon/human), "icon-x=15;icon-y=18;vis-x=15;v...")
 - the chainsaw (Wielded) (/obj/item/chainsaw): melee attack chain([REDACTED_MOB] (/mob/living/carbon/human), [REDACTED_MOB_SECOND] (/mob/living/carbon/human), "icon-x=15;icon-y=18;vis-x=15;v...")
 - [REDACTED_MOB] (/mob/living/carbon/human): ClickOn([REDACTED_MOB_SECOND] (/mob/living/carbon/human), "icon-x=15;icon-y=18;vis-x=15;v...")
 - [REDACTED_MOB_SECOND] (/mob/living/carbon/human): Click(the floor (75,171,2) (/turf/open/floor/wood), "mapwindow.map", "icon-x=15;icon-y=18;vis-x=15;v...")
 - [REDACTED_CLI] (/client): Click([REDACTED_MOB_SECOND] (/mob/living/carbon/human), the floor (75,171,2) (/turf/open/floor/wood), "mapwindow.map", "icon-x=15;icon-y=18;vis-x=15;v...")
 - 
```

### PR DESCRIPTION
**Pull request by CydiaLamiales**
## About The Pull Request
Fixes the divisions by 0
* closes https://github.com/BeeStation/BeeStation-Hornet/issues/9449
* closes https://github.com/BeeStation/BeeStation-Hornet/issues/9448
* closes https://github.com/BeeStation/BeeStation-Hornet/issues/9447
* closes https://github.com/BeeStation/BeeStation-Hornet/issues/9445
* closes https://github.com/BeeStation/BeeStation-Hornet/issues/9443
* closes https://github.com/BeeStation/BeeStation-Hornet/issues/9442

## Why It's Good For The Game
Bugfix good

## Testing Photographs and Procedure

<details>
<summary>Screenshots&Videos</summary>

I dunno how to prove that I fixed these as I couldn't even recreate the original divisions by 0 on local.

</details>

## Changelog
:cl:
fix: Fixed some divisions by 0
/:cl:

### CODE REVIEW

File: code/modules/projectiles/guns/energy.dm [hunk 1]
```diff
@@ -75,6 +75,8 @@
 
 	if(shot.e_cost > 0)
 		shot_cost_percent = FLOOR(clamp(shot.e_cost / cell.maxcharge, 0.01, 1) * 100, 1)
+		if(shot_cost_percent <= 1) //Division by 0 protection
+			shot_cost_percent = 1
 		max_shots = round(100/shot_cost_percent)
 		shots_left = round(batt_percent/shot_cost_percent)
```
Comment (EvilDragonfiend) [inline code/modules/projectiles/guns/energy.dm]:
```suggestion
		max_shots = shot_cost_percent ? round(100/shot_cost_percent) : 0 //Division by 0 protection
		shots_left = shot_cost_percent  ? round(batt_percent/shot_cost_percent) : 0 //Division by 0 protection
```

File: code/modules/power/cell.dm [hunk 2]
```diff
@@ -75,7 +75,10 @@
 		add_overlay("cell-o1")
 
 /obj/item/stock_parts/cell/proc/percent()		// return % charge of cell
-	return 100*charge/maxcharge
+	var/current_maxcharge = maxcharge
+	if(current_maxcharge <= 1) //Division by 0 protection
+		current_maxcharge = 1
+	return 100 * charge / current_maxcharge
```
Comment (EvilDragonfiend) [inline code/modules/power/cell.dm]:
```suggestion
	return current_maxcharge ? 100 * charge / current_maxcharge : 0 //Division by 0 protection
```

File: code/game/data_huds.dm [hunk 3]
```diff
@@ -408,6 +408,9 @@
 	var/icon/I = icon(icon, icon_state, dir)
 	holder.pixel_y = I.Height() - world.icon_size
 	if(cell)
+		var/maxcharge = cell.maxcharge
+		if(maxcharge <= 1) //Division by 0 protection
+			maxcharge = 1
 		var/chargelvl = cell.charge/cell.maxcharge
```
Comment (EvilDragonfiend) [inline code/game/data_huds.dm]:
```suggestion
		var/chargelvl = cell.maxcharge ? cell.charge/cell.maxcharge : 0 //Division by 0 protection
```

**Review by EvilDragonfiend — CHANGES_REQUESTED** [review]

**Comment by EvilDragonfiend** [comment]
This PR doesn't include the fixes of:
* #9444
* #9442

**Comment by CydiaLamiales** [comment]
> This PR doesn't include the fixes of:
> 
> * [Division by zero - temperature_pump.dm,56 #9444](https://github.com/BeeStation/BeeStation-Hornet/issues/9444)
> * [Division by zero - data_huds.dm,411 #9442](https://github.com/BeeStation/BeeStation-Hornet/issues/9442)

Whoops, it doesn't include the fix for the temperature pump
but it does have the fix for the data huds

**Comment by EvilDragonfiend** [comment]
image

mecha.dm has the same issue here

**Review by EvilDragonfiend — APPROVED** [review]
all looks good

File: code/datums/components/butchering.dm [hunk 4]
```diff
@@ -53,15 +53,18 @@
 					"<span class = 'userdanger'>Something is cutting into your neck!</span>", NONE)
 
 	playsound(H.loc, butcher_sound, 50, TRUE, -1)
-	if(do_after(user, CLAMP(500 / source.force, 30, 100), H) && H.Adjacent(source))
+	var/item_force = source.force
+	if(item_force <= 1) //Division by 0 protection
```
Comment (Crossedfall) [inline code/datums/components/butchering.dm]:
```suggestion
	if(!item_force) //Division by 0 protection
```

File: code/modules/admin/battle_royale.dm [hunk 5]
```diff
@@ -246,6 +246,8 @@ GLOBAL_DATUM(battle_royale, /datum/battle_royale_controller)
 		return
 	//Once every 15 seconsd
 	// 1,920 seconds (about 32 minutes per game)
+	if(field_delay <= 0) //Division by 0 protection
```
Comment (Crossedfall) [inline code/modules/admin/battle_royale.dm]:
```suggestion
	if(!field_delay) //Division by 0 protection
```

File: code/modules/events/spacevine.dm [hunk 6]
```diff
@@ -426,6 +426,8 @@
 	if(production != null && production <= 10) //Prevents runtime in case production is set to 11.
 		spread_cap *= (11 - production) / 5 //Best production speed of 1 doubles spread_cap to 60 while worst speed of 10 lowers it to 6. Even distribution.
 		spread_multiplier /= (11 - production) / 5
+		if(spread_multiplier <= 0) // Division by 0 protection
```
Comment (Crossedfall) [inline code/modules/events/spacevine.dm]:
```suggestion
		if(!spread_multiplier) // Division by 0 protection
```

**Pull request merged**